src/corecel/data/Copier.hh

```diff
@@ -39,6 +40,8 @@ struct Copier
     static constexpr auto dstmem = M;
 
     inline void operator()(MemSpace srcmem, Span<T const> src) const;
+    inline void
+    operator()(MemSpace srcmem, Span<T const> src, StreamId stream) const;
```

Comment:
Since the stream will presumably be the same for all the copy operations made by a single Copier instance, it should go in the constructor rather than call operator.

Reply:
Yeah, wasn't sure I ended up going for specifying it in the call operator because in the constructor we have to check at runtime which version to use and store an extra id. Pretty minimal overhead but nothing would technically prevent to reuse the same Copier on different streams.

Reply:
Yeah, it makes sense when you put it that way. But presumably since the destination is the same, if you're pinning it then it's async and on the same stream; if it's not, then you're doing synchronized anyway.